pkg/kubelet/lifecycle/predicate.go

```diff
@@ -236,9 +234,6 @@ func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]Pr
 		})
 	}
 
-	if !pluginhelper.PodMatchesNodeSelectorAndAffinityTerms(pod, nodeInfo.Node()) {
```

Comment:
Maybe we need to keep the nodeselector check.

Reply:
according to https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity, nodeSelector is like requiredDuringSchedulingIgnoredDuringExecution